app/components/markdown/markdown_latex_block/index.tsx

```diff
@@ -0,0 +1,232 @@
+        const title = intl.formatMessage({
+            id: 'mobile.routes.code',
+            defaultMessage: '{language} Code',
+        }, {
+            language: languageDisplayName,
+        });
+
+        Keyboard.dismiss();
+        requestAnimationFrame(() => {
+            goToScreen(screen, title, passProps);
+        });
+    });
+
+    const handleLongPress = useCallback(() => {
+        if (managedConfig?.copyAndPasteProtection !== 'true') {
+            const renderContent = () => {
+                return (
+                    <View
+                        testID='at_mention.bottom_sheet'
+                        style={styles.bottomSheet}
+                    >
+                        <SlideUpPanelItem
+                            icon='content-copy'
+                            onPress={() => {
+                                dismissBottomSheet();
+                                Clipboard.setString(content);
+                            }}
+                            testID='at_mention.bottom_sheet.copy_code'
+                            text={intl.formatMessage({id: 'mobile.markdown.code.copy_code', defaultMessage: 'Copy Code'})}
+                        />
+                        <SlideUpPanelItem
+                            destructive={true}
+                            icon='cancel'
+                            onPress={() => {
+                                dismissBottomSheet();
+                            }}
+                            testID='at_mention.bottom_sheet.cancel'
+                            text={intl.formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'})}
+                        />
+                    </View>
+                );
+            };
+
+            bottomSheet({
+                closeButtonId: 'close-code-block',
+                renderContent,
+                snapPoints: [bottomSheetSnapPoint(2, ITEM_HEIGHT, insets.bottom), 10],
+                title: intl.formatMessage({id: 'post.options.title', defaultMessage: 'Options'}),
+                theme,
+            });
+        }
+    }, [managedConfig, intl, insets, theme]);
+
+    const splitContent = () => {
+        const lines = splitLatexCodeInLines(content);
+        const numberOfLines = lines.length;
+
+        if (numberOfLines > MAX_LINES) {
+            return {
+                content: lines.slice(0, MAX_LINES),
+                numberOfLines,
+            };
+        }
+
+        return {
+            lines,
+            numberOfLines,
+        };
+    };
+
+    const onErrorMessage = (errorMsg: Error) => {
+        return <Text style={styles.errorText}>{'Error: ' + errorMsg.message}</Text>;
+    };
+
+    const onRenderErrorMessage = ({error}: {error: Error}) => {
+        return <Text style={styles.errorText}>{'Render error: ' + error.message}</Text>;
+    };
+
+    const languageDisplayName = getHighlightLanguageName('latex');
+    const {lines, numberOfLines} = splitContent();
```

Comment:
This seems a costly enough operation to consider useMemo. This is, of course, depending on how often do we expect this component to re-render.